exp/api/v1alpha3/conversion.go

```diff
@@ -17,26 +17,72 @@ limitations under the License.
 package v1alpha3
 
 import (
-	"k8s.io/apimachinery/pkg/conversion"
-	v1beta1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	apimachineryconversion "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // Convert_v1alpha3_MachinePoolSpec_To_v1beta1_MachinePoolSpec is an autogenerated conversion function.
-func Convert_v1alpha3_MachinePoolSpec_To_v1beta1_MachinePoolSpec(in *MachinePoolSpec, out *v1beta1.MachinePoolSpec, s conversion.Scope) error {
+func Convert_v1alpha3_MachinePoolSpec_To_v1beta1_MachinePoolSpec(in *MachinePoolSpec, out *v1beta1.MachinePoolSpec, s apimachineryconversion.Scope) error {
 	return autoConvert_v1alpha3_MachinePoolSpec_To_v1beta1_MachinePoolSpec(in, out, s)
 }
 
-func Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in *MachinePool, out *v1beta1.MachinePool, s conversion.Scope) error {
+func Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in *MachinePool, out *v1beta1.MachinePool, s apimachineryconversion.Scope) error {
 	if err := autoConvert_v1alpha3_MachinePool_To_v1beta1_MachinePool(in, out, s); err != nil {
 		return err
 	}
 
 	// Replace v1alpha3 finalizer to allow old MachinePools to get deleted.
-	if !controllerutil.ContainsFinalizer(out, MachinePoolFinalizer) {
+	if controllerutil.ContainsFinalizer(out, MachinePoolFinalizer) {
 		controllerutil.RemoveFinalizer(out, MachinePoolFinalizer)
 		controllerutil.AddFinalizer(out, v1beta1.MachinePoolFinalizer)
 	}
 
 	return nil
 }
+
+func (src *MachinePool) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta1.MachinePool)
+
+	if err := Convert_v1alpha3_MachinePool_To_v1beta1_MachinePool(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Preserve Hub data on conversion except for metadata
+	if err := utilconversion.MarshalData(src, dst); err != nil {
```

Comment:
Not sure if the annotation is meant to be used for conversions in this "direction" (spoke=>hub). 
In all other cases we currently have we implement it for hub=>spoke (i.e. in ConvertFrom)

I think this should work as long as we don't try to do it for the same resource in both directions at the same time.

Reply:
@vincepri WDYT?

Reply:
Talked to vince. We should never add the conversion data / annotation in this direction because then it's stored in etcd, which is something we don't want.